### PR TITLE
Basic user interaction

### PR DESCRIPTION
**Pull request by tacodtripe**
In this branch, we create a mock flow from a tic-tac-toe game to inspect the logic involved in the game.

### CODE REVIEW

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great work @team :clap: :clap: However, some requirements need to be implemented in order to get approval for this milestone. Check :arrow_down: 

#### Behavior 
* [ ] In this step, not everything must work, the point is to make sure that you thought about everything listed below. You might do it only with inline comments. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/logic.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
* [ ] Basic game flow is prepared: ([screenshot hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/logic.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
  * [ ] `[REQUIRED]` game tells which player turn it is
  * [ ] `[REQUIRED]` game asks the player to select from available moves
  * [x] `[REQUIRED]` game informs player if the selected move is invalid
  * [ ] `[REQUIRED]` game displays board after the player move
  * [x] `[REQUIRED]` game informs player if the selected move is a winning move
  * [x] `[REQUIRED]` game informs player if the selected move is a draw move
  * [ ] `[REQUIRED]` game repeats all actions for the next player's move

##### Implementation details

* [ ] A loop is needed in this game (recommended is `while` but anything else will do) ([OdinProject hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/loop.png))
* [ ] Inside the loop, all actions above should be repeated for each player separately (i,e one player turn per iteration).
* [ ] A loop needs to be terminated at some point (after a winning move or draw to be more specific). It can be done with variable:

```ruby
game_on = true
# initial instructions for player
current_player = player_one

while game_on
  # loop for each move
  # actions mentioned in the behaviour list above
  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  else
  # switch current player
  end
end
```

### That's it :relieved: When finished, kindly submit a code re-review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

File: bin/main [hunk 1]
```diff
@@ -1,3 +1,61 @@
...
+puts '---------'
+puts '4 | 5 | 6'
+puts '---------'
+puts '7 | 8 | 9'
+
+puts "\nPlayer 1 (X), insert your name:"
+player_1 = gets.chomp # rubocop:todo Naming/VariableNumber
+
+puts "\nPlayer 2 (O), insert your name:"
+player_2 = gets.chomp # rubocop:todo Naming/VariableNumber
```
Comment (stratospherique) [inline bin/main]: - Adding these rubocop rules could have been avoided if you name the player variable as 'player_one'. :pray:  remove these rules

File: bin/main [hunk 2]
```diff
@@ -1,3 +1,61 @@
...
+
+puts "\n#{player_1} make your move"
+gets.chomp
+
+game_on = false
+# initial instructions for player
+current_player = 1
```
Comment (stratospherique) [inline bin/main]: - current_player is supposed to host the instance of the current player(Object), not an number :mag:

File: bin/main [hunk 3]
```diff
@@ -1,3 +1,61 @@
...
+puts 'Your move has been displayed in the board'
+# we need to check if the game has been defined as a tie or anyone won,if not it's player 2 turn.
+
+puts "\n#{player_2} make your move"
+gets.chomp
+puts 'now we will verify your selection'
+puts 'Your move has been displayed in the board'
+# the game will loop until it is defined
+
+puts "\nCongrats #{player_2} you won!"
```
Comment (stratospherique) [inline bin/main]: - The part is supposed to be removed because it was replaced by the loop above.

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great work implementing the suggested changes :clap: :clap: , however, small changes need to be applied in order to get approval for this milestone. Check :arrow_down: 

- **[REQUIRED]** Check the code comments below, they should help you refactor your project.

### That's it :relieved: When finished, kindly submit a code review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hi Team 👋

You've obviously both done well on this.🚀
Let's make the following changes to make it better.

- [ ] Somewhere within the loop you should terminate the game when there is a winner/loser/tie. Kindly add a condition that ends the game based on these conditions.

Kindly fix this issue and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

**Comment by lekegitrepo** [comment]
### Hi @tacodtripe and @Morrism1 :clap: :clap:
#### Good so far. Some requirements are not well implemented, check my comment below :point_down:

**_1. The game did not check for an invalid move_** :point_down: 
Screenshot from 2020-06-01 05-55-40

**_2. After two rounds the game terminals and declares a winner_**
**_which means the full implementation of these requirements :point_down:  are missing:_**
* [ ] `[REQUIRED]` game informs player if the selected move is a winning move
* [ ] `[REQUIRED]` game informs player if the selected move is a draw move

**_3. The board did not display after the player move_**

**_fix these and submit for review_**

### Reviewed by Temitayo Adeleke.

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
#### Status: Request changes

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hello again Team 👋

Great job working with previous reviews, we are just a step away, hang in there.💪🚀

- [ ] Let's make this addition, we want the player to see the board before and after each move. Would you agree that helps the player know which board positions are filled? However, you can consider using an inline comment to show this message in the game loop.

Kindly fix this issue and request another review👍
Feel free to contact me on slack(michgolden) if you need further clarification.

HAPPY CODING👨‍💻

**Review by mikenath223 — APPROVED** [review]
### CONGRATULATIONS  TEAM 👏👏👏

This project meets the requirements for approval.

**STATUS: APPROVED ✅**


Don't forget to fill the project completion form👍
Good luck with the next project.

HAPPY CODING👨‍💻

**Pull request merged**